design-proposals/edge-infra-day2-improvements.md

```diff
@@ -0,0 +1,357 @@
+    tell anything about the fact that the update will be done in the next 
+    scheduled maintenance. The field is a string.
+    - mutable: list of packages ready to be upgraded on the Edge Node, 
+      empty if no new versions are available.
+    - immutable: latest EMT version, empty if the version is already latest.
+  - `next_update_policy`: The update policy that is applied during the 
+    maintenance schedule.
+  
+New Resources:
+
+- **OS Update Policy**: defines the Policy that must be applied during an 
+  update. This resource can be linked to multiple Instances, Regions and or 
+  Sites. Some fields and edges are valid for Mutable or Immutable OSes.
+  - `install_packages`: list of packages to be installed during the update. 
+    This field is used only for Mutable OSes.
+  - `update_sources`: 'DEB822 Source Format' for Debian style OSs, contains 
+    the sources of the packages to be installed during the update, if new 
+    sources needs to be added.
+  - `update_policy`: enum field, defines which policy to use:
+    - `UPDATE_POLICY_LATEST`: upgrade to latest version. 
+      - mutable: upgrade all packages to latest versions.
+      - immutable: upgrade the OS version to the latest.
+    - `UPDATE_POLICY_TARGET`: upgrade to a specific version.
+      - mutable: upgrade all packages to the specified version defined in 
+        the `install_package` list.
+      - immutable: upgrade the OS version to OS Version linked by the 
+        `target_os` edge.
+  - `target_os`: edge to the OS Profile that must be installed during the 
+    update. This field is used only for Immutable OSes.
+
+- **OS Update Job**: stores the information about the update job that run 
+  on the Edge Node. This resource is created when the update is started, 
+  and stores information about what happened in the Edge Node during the 
+  Update.
+  - `status`: stores the short description of what happened during the update.
+  - `status_indicator`: enum field, moder status indicator:
+    - `STATUS_INDICATION_ERROR`: an error happened during the update.
+    - `STATUS_INDICATION_IN_PROGRESS`: unused, the progress about the 
+      update is kept into the `update_status` field of the Instance.
+    - `STATUS_INDICATION_IDLE`: the update was successful.
+  - `status_timestamp`: timestamp of the last status update. Overlaps with 
+    the `end_time`, but it's kept for consistency with Modern Status. Also, 
+    could be used in the future, if we deprecate the Instance `update_status`.
+  - `status_details`: detailed description of what happend during the 
+    update. For example, the list of packages that were installed or 
+    updated, or the reason why the Immutable OS update failed.
+  - `start_time`: timestamp of when the update job started.
+  - `end_time`: timestamp of when the update job ended.
+  - `applied_policy`: the policy that was applied during the update. 
+    Provides information of what was the target Update Policy for this update.
```

Comment:
Should this be linked to the schedule resource as well?  Would allow the user to determine what schedule triggered the update.

Reply:
I wanted to keep the resource lighter and avoid that link (potentially also the link to the `applied_policy` could be remove). Also, it would be difficult to understand which actually resource triggered the update. Multiple could apply.